File: Algorithms/763. Partition Labels/php/solution.php
<?php
class Solution
{

    /**
     * @param String $s
     * @return Integer[]
     */
    function partitionLabels($s)
    {
        $len = strlen($s);
        $last = [];
        $ans = [];
        for ($i = 0; $i < $len; $i++) {
            $last[$s[$i]] = $i;
        }
        $anchor = -1;
        $j = 0;
        for ($i = 0; $i < $len; $i++) {
            $j = max($j, $last[$s[$i]]);
            if ($i == $j) {
                array_push($ans, $j - $anchor);
                $anchor = $j;
            }
        }
        return $ans;
    }
}

File: Algorithms/763. Partition Labels/swift/solution.swift
class Solution {
    func partitionLabels(_ s: String) -> [Int] {
        var last = [Character:Int]()
        for (i, c) in s.enumerated() {
            last[c, default: 0] = i
        }

        var ans = [Int]()
        var j = 0
        var anchor = -1
        for (i, c) in s.enumerated() {
            j = max(j, last[c, default: 0])
            if i == j {
                ans.append(i - anchor)
                anchor = i
            }
        }
        return ans
    }
}
